agent/agent_endpoint.go

```diff
@@ -944,8 +944,11 @@ func (s *HTTPServer) AgentConnectCALeafCert(resp http.ResponseWriter, req *http.
 		Service: serviceName, // Need name not ID
 	}
 	var qOpts structs.QueryOptions
+
 	// Store DC in the ConnectCALeafRequest but query opts separately
-	if done := s.parse(resp, req, &args.Datacenter, &qOpts); done {
+	s.parseDC(req, &args.Datacenter)
+	s.parseTokenProxyResolve(req, &qOpts.Token, false)
```

Comment:
It's a bit janky that when we want to use the extended verification the boiler plate increases.

I realise this probably had the advantage of being a tiny change compared with refactoring verify method a bit but I feel like that might be worth it.

I'm in two minds about whether to untangle this now or just land it as a fix and come back later.

I think it is important enough to fix nicely though? What do you think?

---

Oh how about:

 - make a `parseWithoutResolvingProxyToken` (yay names) that does all of the same as `parse` but with false here.

Or alternatively, leave this just as `parse` as it is here, but change `verifyProxyToken` to accept the `req` directly and do it's own unpacking of the raw value rather than relying on `qOpts`?